crates/circuit/src/operations.rs

```diff
@@ -374,6 +382,23 @@ impl Operation for StandardGate {
                 }
                 _ => None,
             },
+            Self::U1Gate => match params[0] {
+                Param::Float(val) => Some(aview2(&gate_matrix::u1_gate(val)).to_owned()),
+                Param::ParameterExpression(_) => None,
```

Comment:
This looks different than other 1-parameter gates above, which have
```
gate => params  {
  [Param::Float(...)] => matrix,
  _ => None,
}
```
why is the additional match with `ParameterExpression` not supported here, but is with the others?

Reply:
The `_ =>` syntax is basically a catch-all match instead of that here I'm just explicitly matching the other two choices. I'll update this to use the catch-all match syntax though, I didn't actually do this with any intention, this PR came from a test branch and I updated it from when #12459 only had 2 variants for the `Param` enum to add the 3rd condition.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/12572/commits/65ff00991a3acd6d3619686a9ced942ebced0662